articles/azure/azs-how-configure-custom-script-extension.md

```diff
@@ -0,0 +1,338 @@
+$CustomScriptStorageAccountName = "<output form="customscriptstorageaccountname" name="result" style="display: inline;">customscript<span id="RandNum"></span></output>".ToLower()
+$FilePath = "<output form="filepath" name="result" style="display: inline;">C:\Users\User1\SetRootPassword.sh</output>"
+$CommandToExecute = "<output form="commandtoexecute" name="result" style="display: inline;">sh SetRootPassword.sh Password123!</output>"
+
+# Create a new storage account
+Write-Output -InputObject "Creating storage account and container"
+$StorageAccount = New-AzureRmStorageAccount -Location $Location -ResourceGroupName $RGName -Type "Standard_LRS" -Name $CustomScriptStorageAccountName
+
+# Get storage account context
+$Context = $StorageAccount.Context
+$Container = New-AzureStorageContainer -Name $ContainerName -Context $Context
+
+# Retrieve storage account key
+$StorageAccountKey = (Get-AzureRmStorageAccountKey -ResourceGroupName $RGName -Name $CustomScriptStorageAccountName).Value[0]
+
+# Retrieve storage blob endpoint
+$ScriptBlobUrl = $Container.Context.BlobEndPoint
+
+# Upload script extension to the storage account
+Write-Output -InputObject "Uploading custom script extension to storage account"
+Set-AzureStorageBlobContent -File $FilePath -Container $ContainerName -Blob $CustomScriptFileName -Context $Context
+
+# Creating script location string
+$ScriptLocation = $ScriptBlobUrl + "$ContainerName/" + $CustomScriptFileName
+
+# Add custom script extension to Linux VM
+Write-Output -InputObject "Adding custom script extension to VM"
+$Extensions = Get-AzureRmVMExtensionImage -Location "frn00006" -PublisherName "Microsoft.Azure.Extensions" -Type "CustomScript"
+$ExtensionVersion = $Extensions[0].Version[0..2] -join ""
+$ScriptSettings = @{"fileUris" = @("$ScriptLocation")};
+$ProtectedSettings = @{"storageAccountName" = $CustomScriptStorageAccountName; "storageAccountKey" = $StorageAccountKey; "commandToExecute" = $CommandToExecute};
+Set-AzureRmVMExtension -ResourceGroupName $RGName -Location $Location -VMName $VMName -Name $Extensions[0].Type -Publisher $Extensions[0].PublisherName -ExtensionType $Extensions[0].Type -TypeHandlerVersion $ExtensionVersion -Settings $ScriptSettings -ProtectedSettings $ProtectedSettings
+</code></pre>
+
+#### File URI
+
+<pre><code class="language-PowerShell">$FileUri = "<output form="fileuri" name="result" style="display: inline;">https://raw.githubusercontent.com/UKCloud/AzureStack/master/Users/Extensions/Linux/SetRootPassword.sh</output>"
+$CommandToExecute = "<output form="commandtoexecute" name="result2" style="display: inline;">sh SetRootPassword.sh Password123!</output>"
+
+# Add custom script extension to Linux VM
+Write-Output -InputObject "Adding custom script extension to VM"
+$Extensions = Get-AzureRmVMExtensionImage -Location "frn00006" -PublisherName "Microsoft.Azure.Extensions" -Type "CustomScript"
```

Comment:
Can we pull the location either directly from the stack or from a variable in the creation script?